File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
public class Solution {
    public int solution(int left, int right) {
        int sum = 0;
        for (int num = left; num <= right; num++) {
            int divisorCount = countDivisors(num);
            if (divisorCount % 2 == 0) {
                sum += num;
            } else {
                sum -= num;
            }
        }
        return sum;
    }

    // 약수의 개수를 세는 함수
    private int countDivisors(int n) {
        int count = 0;
        for (int i = 1; i <= n; i++) {
            if (n % i == 0) {
                count++;
            }
        }
        return count;
    }
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
function solution(left, right) {
    var answer = 0;
    
    for(let i =left; i<=right; i++){
        if(numCount(i)%2===0){
            answer +=i
        }else{
            answer -=i
        }
    }
    
    
    return answer;
}

function numCount (n){
        let count = 0
        for(let i=1; i<n+1; i++){
            if(n%i===0){
                count++
            }
    }
        return count;
}